server/mailet/mailets/src/main/java/org/apache/james/transport/matchers/dlp/DlpDomainRules.java

```diff
@@ -106,16 +106,22 @@ public boolean doTest(Mail mail) throws MessagingException, IOException {
                 if (content instanceof String) {
                     return Stream.of((String) content);
                 }
+
+                return extractContentsComplexType(content)
```

Comment:
First, I'm not sure what is tail recursive ...
Second, AFAIK Java doesn't support tail recursion optimization ...